Fix duplicate note on internal feature gates with associated issues

gurry opened the pull request:
Fixes #116293

Note sure if I should add tests because the issue occurs only for feature gates having associated issues and that set of feature gates will change unpredictably leading to an unnecessary churn in tests.

petrochenkov commented on compiler/rustc_lint/src/builtin.rs:
```diff
@@ -2273,7 +2273,7 @@ impl EarlyLintPass for IncompleteInternalFeatures {
                         INTERNAL_FEATURES,
                         span,
-                        BuiltinInternalFeatures { name, note },
+                        BuiltinInternalFeatures { name, note: None }, // Passing 'None' as 'BuiltinInternalFeatures' struct already has '#[note]' attr
```
```suggestion
                        BuiltinInternalFeatures { name },
```
If `note` is always `None`, then the corresponding field should be removed from the structure.

gurry replied:
Thanks. Done.

gurry committed:
Fix duplicate note on internal feature gate

The BuiltinInternalFeatures gate already has a struct level #[note]
attribute. The additional note field in it caused a duplicate to be
displayed when it was set to Some(...) which happened when the
feature had an associated issue

gurry commented:
@rustbot label -S-waiting-on-author +S-waiting-on-review

petrochenkov commented:
Thanks!
@bors r+ rollup

bors commented:
:pushpin: Commit 66e643d0f8a6406525ecbf61f8d99005208f97f4 has been approved by `petrochenkov`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=petrochenkov 66e643d0f8a6406525ecbf61f8d99005208f97f4 -->
<!-- homu: {"type":"Approved","sha":"66e643d0f8a6406525ecbf61f8d99005208f97f4","approver":"petrochenkov","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/6492875403/job/17636451320)
Approved by: petrochenkov
Pushing 4f05e9545210b7cf4db544a5aa2128b9ce0eb894 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"petrochenkov","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/6492875403/job/17636451320"},"merge_sha":"4f05e9545210b7cf4db544a5aa2128b9ce0eb894"} -->

rust-timer commented:
Finished benchmarking commit (4f05e9545210b7cf4db544a5aa2128b9ce0eb894): [comparison URL](https://perf.rust-lang.org/compare.html?start=3d575a2f2ef8a6eb99064bb31c16feb8d508f1ee&end=4f05e9545210b7cf4db544a5aa2128b9ce0eb894&stat=instructions:u).

### Overall result: ❌ regressions - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=3d575a2f2ef8a6eb99064bb31c16feb8d508f1ee&end=4f05e9545210b7cf4db544a5aa2128b9ce0eb894&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 0.4% | [0.4%, 0.4%] | 3     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=3d575a2f2ef8a6eb99064bb31c16feb8d508f1ee&end=4f05e9545210b7cf4db544a5aa2128b9ce0eb894&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 3.7% | [3.7%, 3.7%] | 1     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=3d575a2f2ef8a6eb99064bb31c16feb8d508f1ee&end=4f05e9545210b7cf4db544a5aa2128b9ce0eb894&stat=cycles:u)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -7.3% | [-7.3%, -7.3%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Binary size](https://perf.rust-lang.org/compare.html?start=3d575a2f2ef8a6eb99064bb31c16feb8d508f1ee&end=4f05e9545210b7cf4db544a5aa2128b9ce0eb894&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 628.393s -> 626.317s (-0.33%)
**Artifact size**: 271.29 MiB -> 271.25 MiB (-0.01%)
